---
title: "Sex_check_removed_samples"
author: "Annika Jorgensen"
date: "2023-09-22"
output: html_document
---
**README**
PURPOSE: Automating the output of gene expression barplots of all ICGC samples for all y-linked genes.

METHODS: A hashmap is created to link the gene name with a corresponding ensembl id. A voom analysis was done all of the expression data and then the data was subsetted by tumor and tumor adjacent and then placed into a bar plot. The bar plots are ordered by viral etiology and sex. The bar plot was also colored by both sex and etiology. 

CONCLUSION: 
**Results**
### **Libraries**
The first chunk of code is dedicated to installing the libraries. These libraries are to help execute the differential analysis and helps visualize the data. The code was not included for concision.  
```{r Libraries, include=FALSE}

library(RColorBrewer)
library(matrixStats)
library(ggplot2)
library(edgeR)
library(DESeq2)
library(limma)
library(doParallel)
library(variancePartition)
library(org.Hs.eg.db)
library(clusterProfiler)
library(GOSemSim)
library(biomaRt)
library(UpSetR)
library(VennDiagram)
library(ggrepel)
library(dplyr)
library(stringr)
library(forcats)
library(stats)
library(gridExtra)

setwd('~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples')
```

**Defining Colors** 

This chunk defines color palette variables that are going to be used in plots later on the script. These variables are defined by converting BrewerCode palettes into palettes that can be used in R. 
```{r Color_Palettes}
viralPalette <- brewer.pal(8, "Set1")
hbvColor <- viralPalette[1]
hcvColor <- viralPalette[2]
bothColor <- viralPalette[3]
neitherColor <- viralPalette[4]

sexTissuePalette <- brewer.pal(12, "Paired")
maleTumorColor <- sexTissuePalette[4]
maleAdjacentColor <- sexTissuePalette[3]
femaleTumorColor <- sexTissuePalette[6]
femaleAdjacentColor <- sexTissuePalette[5]
```

**Read in data**

This code is where you read in all the data files that are going to be used in the script. The data is also converted into a variety of variables that makes the data easier to handle. The data is also cleaned up to make sure the analysis done later is accurate and precise. 
```{r Reading_in_data}
metadata <- read.table("~/Desktop/ResearchProjects/LiverCancer/Metadata/metadata_for_de.csv", row.names=1,header=TRUE, sep=",") #changing the name of the file
tumorAdjacentExp <- read.table("~/Desktop/ResearchProjects/LiverCancer/Metadata/japan_all_samples_salmon_expression_counts.txt", row.names = 1, header=TRUE) #changing the name of the file 
colnames(tumorAdjacentExp) <- gsub("\\.", "-", colnames(tumorAdjacentExp)) #changing the column names 

# Importing gene annotations
#genes <- read.table("gencode.v25.chr_patch_hapl_scaff.annotation.bed", header=FALSE, sep="\t")
genes <- read.table("~/Desktop/ResearchProjects/LiverCancer/Metadata/gencodeTranscripts.txt", header=TRUE, sep="\t")
genes <- data.frame(genes)
tumorAdjacentExp <- tumorAdjacentExp[rownames(tumorAdjacentExp) %in% genes$GENEID ,]
genes <- genes[match(rownames(tumorAdjacentExp), genes$GENEID),]
# Calculating gene length, this is needed for calculating the FPKM values
genes$length <- with(genes, end - start)

# Removing Samples due to low quality
metadata <- metadata[!(metadata$ID == "RK023") , ]
metadata <- metadata[!(metadata$ID == "RK106") , ]
metadata <- metadata[!(metadata$ID == "RK113") , ]
metadata <- metadata[!(metadata$ID == "RK135") , ]
metadata <- metadata[!(metadata$ID == "RK105") , ]
metadata <- metadata[!(metadata$ID == "RK116") , ]
metadata <- metadata[!(metadata$ID == "RK066") , ]
metadata <- metadata[!(metadata$ID == "RK096") , ]

#Removing both and NBNC samples
metadata <- metadata[!(metadata$Virus_infection == "NBNC"), ]
metadata <- metadata[!(metadata$Virus_infection == "both"), ]

# Subsetting and ordering metadata to match the count matrix
tumorAdjacentExpSubset <- tumorAdjacentExp[,colnames(tumorAdjacentExp) %in% metadata$sampleid]
metadataSubset <- metadata[metadata$sampleid %in% colnames(tumorAdjacentExpSubset),]
metadataSubset <- metadataSubset[match(colnames(tumorAdjacentExpSubset), metadataSubset$sampleid),]
rownames(metadataSubset) <- metadataSubset$sampleid

# Adding tissue type, converting categorical variables to factors
metadataSubset$tumor <- as.numeric(grepl('tumor', metadataSubset$sampleid, ignore.case=T))

#Swapping lesion type for sample RK169
metadataSubset["RK169-tumor-XY","tumor"] <- 0
metadataSubset["RK169-adjacent-XY","tumor"] <- 1

#Changing rownames to match swapped lesion type
rownames(metadataSubset)[rownames(metadataSubset)=="RK169-tumor-XY"] <- "RK169_adjacent-XY"
rownames(metadataSubset)[rownames(metadataSubset)=="RK169-adjacent-XY"] <- "RK169_tumor-XY"
rownames(metadataSubset)[rownames(metadataSubset)=="RK169_adjacent-XY"] <- "RK169-adjacent-XY"

rownames(tumorAdjacentExpSubset)[rownames(tumorAdjacentExpSubset)=="RK169-tumor-XY"] <- "RK169_adjacent-XY"
rownames(tumorAdjacentExpSubset)[rownames(tumorAdjacentExpSubset)=="RK169-adjacent-XY"] <- "RK169_tumor-XY"
rownames(tumorAdjacentExpSubset)[rownames(tumorAdjacentExpSubset)=="RK169_adjacent-XY"] <- "RK169-adjacent-XY"

#Swapping lesion type for sample RK065
metadataSubset["RK065-tumor-XX","tumor"] <- 0
metadataSubset["RK065-adjacent-XX","tumor"] <- 1

#Changing rownames in metadata to match swapped lesion type
rownames(metadataSubset)[rownames(metadataSubset)=="RK065-tumor-XY"] <- "RK065_adjacent-XY"
rownames(metadataSubset)[rownames(metadataSubset)=="RK065-adjacent-XY"] <- "RK065_tumor-XY"
rownames(metadataSubset)[rownames(metadataSubset)=="RK065_adjacent-XY"] <- "RK065-adjacent-XY"

rownames(tumorAdjacentExpSubset)[rownames(tumorAdjacentExpSubset)=="RK065-tumor-XY"] <- "RK065_adjacent-XY"
rownames(tumorAdjacentExpSubset)[rownames(tumorAdjacentExpSubset)=="RK065-adjacent-XY"] <- "RK065_tumor-XY"
rownames(tumorAdjacentExpSubset)[rownames(tumorAdjacentExpSubset)=="RK065_adjacent-XY"] <- "RK065-adjacent-XY"

metadataSubset$gender_tissue <- paste(metadataSubset$Gender, metadataSubset$tumor, sep="_")
metadataSubset$gender_tissue_viral <- paste(metadataSubset$gender_tissue, metadataSubset$Virus_infection, sep="_")
metadataSubset$library_type <- metadataSubset$strandedness
metadataSubset$library_type <- factor(metadataSubset$library_type)
metadataSubset$tumor <- factor(metadataSubset$tumor)
metadataSubset$Ta <- factor(metadataSubset$Ta)
metadataSubset$Portal_vein_invasion <- factor(metadataSubset$Portal_vein_invasion)
metadataSubset$Hepatic_vein_invasion <- factor(metadataSubset$Hepatic_vein_invasion)
metadataSubset$Bile_duct_invasion <- factor(metadataSubset$Bile_duct_invasion)
metadataSubset$Liver_fibrosisc <- factor(metadataSubset$Liver_fibrosisc)
metadataSubset$Prognosis <- factor(metadataSubset$Prognosis)
```

```{r DGE Object}
# Creating the DGEList object
dge <- DGEList(counts=tumorAdjacentExpSubset, genes=genes)
colnames(dge) <- colnames(tumorAdjacentExpSubset)
dge$samples$sex <- metadataSubset$Gender
dge$samples$viral <- factor(metadataSubset$Virus_infection)
dge$samples$ID <- metadataSubset$ID
dge$samples$tumor <- metadataSubset$tumor
dge$samples$gender_tissue <- metadataSubset$gender_tissue
dge$samples$gender_tissue_viral <- metadataSubset$gender_tissue_viral
dge$samples$library_type <- metadataSubset$library_type
dge$samples$edmonson_grade <- metadataSubset$Edmondson_grade
dge$samples$Ta <- metadataSubset$Ta
dge$samples$survival <- metadataSubset$Overall_survival_month

# Inspecting the N of samples in each group
table(dge$samples$gender_tissue_viral)
```

```{r fpkm}
# ======================================
# Filtering expression data
# ======================================

# Keeping genes that have a mean FPKM of at least 0.5 in at least one of the
# groups under investigation and at least 6 reads in at least 10 samples
fpkm <- rpkm(dge, gene.length=dge$genes$length)

M_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HBV")],1,mean,na.rm=TRUE)
M_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HBV")],1,mean,na.rm=TRUE)
M_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HCV")],1,mean,na.rm=TRUE)
M_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HCV")],1,mean,na.rm=TRUE)

F_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HBV")],1,mean,na.rm=TRUE)
F_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HBV")],1,mean,na.rm=TRUE)
F_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HCV")],1,mean,na.rm=TRUE)
F_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HCV")],1,mean,na.rm=TRUE)

keep <- (M_1_HBV_mean_fpkm > 0.5 | M_0_HBV_mean_fpkm > 0.5 | 
           M_1_HCV_mean_fpkm > 0.5 | M_0_HCV_mean_fpkm > 0.5 |
           F_1_HBV_mean_fpkm > 0.5 | F_0_HBV_mean_fpkm > 0.5 |
           F_1_HCV_mean_fpkm > 0.5 | F_0_HCV_mean_fpkm > 0.5 )

dge <- dge[keep,,keep.lib.sizes=FALSE]
dge <- calcNormFactors(dge, method="TMM")
keep <- rowSums(dge$counts > 6) >= 10
dge <- dge[keep,,keep.lib.size=FALSE]
dge <- calcNormFactors(dge, method="TMM")

# N of genes retained after filtering
dim(dge$genes)
```

```{r Voom}
# ===================================== ======================
# ===========================================================
# Analysis of all tumor vs. tumor-adjacent regardless of sex
# ===========================================================
# ===========================================================

# Creating a new design model matrix with the variable of interest and the
# library type
design <- model.matrix(~0+dge$samples$tumor+dge$samples$library_type+dge$samples$Ta)
colnames(design) <- gsub("dge\\$samples\\$tumor", "tumor", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$library_typeunstranded", "library_type", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta2", "Ta2", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta3", "Ta3", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta4", "Ta4", colnames(design))
head(design)

# Running voom again with the new design matrix.
v <- voomWithQualityWeights(dge, design, plot=TRUE)
```

```{r Librarytype}
# Removing batch effects
vCorrectLibtype <- removeBatchEffect(v, batch=v$targets$library_type)

# Exponentiating Expression data
vCorrectLibtype <- 2 ^ vCorrectLibtype
vCorrectLibtype <- as.data.frame(vCorrectLibtype)
```

```{r ChrY}
#installing hash library 
library(hash)

#Storing hashmap as variable
h <- hash()

#Pulling viral etiology designations from metadata
viralType <- metadataSubset$Virus_infection
viralType <- as.data.frame(viralType)

#Renaming the column
colnames(viralType) <- c("Virus")

#Pulling out genes on y chromosome from reference genome
chrY_gene_ids <- genes[which(genes$chr== "chrY"), ]

#Creating a hashmap that links the ensembl id to gene name so the value is the ensembl id and the key is the gene name (if I ask for gene name it will report the ensembl id)
for (idx in 1:dim(chrY_gene_ids)[1]) {
     name <- chrY_gene_ids$gene_name[idx]
     ensembl_id <- chrY_gene_ids$GENEID[idx]
  if (ensembl_id %in% rownames(vCorrectLibtype)) {   
       h[[name]] <- ensembl_id
  }
}

#vCorrectLibtype[grep(pattern="^ENSG00000229807", rownames(vCorrectLibtype)), ]
for (gene in keys(h)) {

  #setting the created hashmap into variable 
  ensembl_id <- h[[gene]]
  
  #Pulling expression data from vCorrectLibtype for the y-linked genes
  expressionData <- vCorrectLibtype[which(rownames(vCorrectLibtype)== ensembl_id),]
  expressionData <- t(expressionData)
  expressionData<- as.data.frame(expressionData)
  
  #Creating dataframe with the gene expression data and corresponding sample viral etiology 
  expressionData <- cbind(expressionData,viralType)
  
  #Males_expressionData <- expressionData[which(grepl("XY",rownames(expressionData),fixed=TRUE)),]
  #Females_expressionData <- expressionData[which(grepl("XX",rownames(expressionData),fixed=TRUE)),]
  
  #Subsetting by tumor samples
  Tumor_expressionData <- expressionData[grepl("tumor",rownames(expressionData),fixed=TRUE),]
  Tumor_expressionData <- as.data.frame(Tumor_expressionData)
  
  #Adding sample ids as rownames to tumor samples
  rownames(Tumor_expressionData) <- rownames(expressionData)[grepl("tumor",rownames(expressionData),fixed=TRUE)]
  
  #Subsetting by adjacent samples 
  Adjacent_expressionData <- expressionData[grepl("adjacent",rownames(expressionData),fixed=TRUE),]
  Adjacent_expressionData <- as.data.frame(Adjacent_expressionData)
  
  #Adding sample ids as rownames to adjacent samples
  rownames(Adjacent_expressionData) <- rownames(expressionData)[grepl("adjacent",rownames(expressionData),fixed=TRUE)]
  
  #Creating new column that adds a 1 if the sample sex is Male and a 0 if the sample is female
  Tumor_expressionData$SexNumeric <- as.numeric(grepl("XY",rownames(Tumor_expressionData),fixed=TRUE))
  Adjacent_expressionData$SexNumeric <- as.numeric(grepl("XY",rownames(Adjacent_expressionData),fixed=TRUE))
  
  #Renaming columns
  colnames(Tumor_expressionData)<- c("Expression","Virus", "SexNumeric")
   
   #Creating a new column that as a 0 if the sample is male HBV, 1 if male HCV, 2 if female HBV, and 3 if female HCV 
  Tumor_expressionData$SexViralNumeric <- ifelse(Tumor_expressionData$SexNumeric=="1" & Tumor_expressionData$Virus=="HBV", 0,
                                                 ifelse(Tumor_expressionData$SexNumeric=="1" & Tumor_expressionData$Virus=="HCV",1,
                                                        ifelse(Tumor_expressionData$SexNumeric=="0" & Tumor_expressionData$Virus=="HBV",2,3)))
  Adjacent_expressionData$SexViralNumeric <- ifelse(Adjacent_expressionData$SexNumeric=="1" & Adjacent_expressionData$Virus=="HBV", 0,
                                                 ifelse(Adjacent_expressionData$SexNumeric=="1" & Adjacent_expressionData$Virus=="HCV",1,
                                                        ifelse(Adjacent_expressionData$SexNumeric=="0" & Adjacent_expressionData$Virus=="HBV",2,3)))
  
  #For loop that adds the sample numbers in a new column called "Names"
  for (i in 1:dim(Tumor_expressionData)[1]) {
    Tumor_expressionData$Names[i] = substr(row.names(Tumor_expressionData)[i],3,5)
  }
  
  for (i in 1:dim(Adjacent_expressionData)[1]) {
    Adjacent_expressionData$Names[i] = substr(row.names(Adjacent_expressionData)[i],3,5)
  }

  #Renaming columns 
   colnames(Tumor_expressionData)<- c("Expression","Virus", "SexNumeric","SexViralNumeric","Names")
   colnames(Adjacent_expressionData)<- c("Expression","Virus", "SexNumeric","SexViralNumeric","Names")
   
  #Creates a box plot of y-linked gene expression data of all tumor samples 
  pdf(paste("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/y_linked_gene_tumor_expression/",gene,"_tumor_expression_data.pdf",sep=""), width=12, height=12)
  print(ggplot(Tumor_expressionData, aes(x=reorder(Names,SexViralNumeric),y= log(as.numeric(Expression)+0.01,10),fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ xlab("Tumor Samples Expression data") + labs(title=paste(gene,"Tumor Samples Expression Data")))
  dev.off() 
   #Creates a box plot of y-linked gene expression data of all tumor adjacent samples
  pdf(paste("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/y_linked_gene_adjacent_expression/",gene,"_tumor_adjacent_expression_data.pdf",sep=""), width=12, height=12)
  print(ggplot(Adjacent_expressionData, aes(x=reorder(Names,SexViralNumeric),y= log(as.numeric(Expression)+0.01,10),fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ xlab("Adjacent Samples Expression data") + labs(title=paste(gene,"Tumor Adjacent Samples Expression Data")))
  dev.off() 
}
```

This next section creates a gene expression plot for XIST with read count data

```{r XIST}
#Pulling viral etiology designations from metadata
viralType <- metadataSubset$Virus_infection
viralType <- as.data.frame(viralType)

#Renaming the column
colnames(viralType) <- c("Virus")

dge <- as.data.frame(dge)
# Pull count data from the dge variable 
XistExpression <- dge[which(dge$gene_name=="XIST"),]
XistExpression<- as.data.frame(XistExpression)

#Removing all columns that are not expression data
XistExpression<- subset(XistExpression, select = -c(chr, start, end, TXNAME, GENEID, length, gene_name))
XistExpression<- t(XistExpression)
XistExpression<- as.data.frame(XistExpression)

#Combining expression data with the corresponding sample viral etiology
XistExpression<- cbind(XistExpression,viralType)
colnames(XistExpression)<- c("Expression","Virus")

#Subsetting by tumor samples
XistTumor <- XistExpression[grepl("tumor",rownames(XistExpression),fixed=TRUE),]
XistTumor <- as.data.frame(XistTumor)

#Subsetting by adjacent samples 
XistAdjacent <- XistExpression[grepl("adjacent",rownames(XistExpression),fixed=TRUE),]
XistAdjacent <- as.data.frame(XistAdjacent)

#Creating new column that adds a 1 if the sample sex is Male and a 0 if the sample is female
XistTumor$SexNumeric <- as.numeric(grepl("XY",rownames(XistTumor),fixed=TRUE))
XistAdjacent$SexNumeric <- as.numeric(grepl("XY",rownames(XistAdjacent),fixed=TRUE))

#Renaming columns
colnames(XistTumor)<- c("Expression","Virus", "SexNumeric")
colnames(XistAdjacent)<- c("Expression","Virus", "SexNumeric")

#Creating a new column that as a 0 if the sample is male HBV, 1 if male HCV, 2 if female HBV, and 3 if female HCV 
  XistTumor$SexViralNumeric <- ifelse(XistTumor$SexNumeric=="1" & XistTumor$Virus=="HBV", 0,
                                                 ifelse(XistTumor$SexNumeric=="1" & XistTumor$Virus=="HCV",1,
                                                        ifelse(XistTumor$SexNumeric=="0" & XistTumor$Virus=="HBV",2,3)))
  
  XistAdjacent$SexViralNumeric <- ifelse(XistAdjacent$SexNumeric=="1" & XistAdjacent$Virus=="HBV", 0,
                                                 ifelse(XistAdjacent$SexNumeric=="1" & XistAdjacent=="HCV",1,
                                                        ifelse(XistAdjacent$SexNumeric=="0" & XistAdjacent$Virus=="HBV",2,3)))
#For loop that adds the sample numbers in a new column called "Names"
for (i in 1:dim(XistTumor)[1]) {
  XistTumor$Names[i] = substr(row.names(XistTumor)[i],3,5)
  }
  
for (i in 1:dim(XistAdjacent)[1]) {
  XistAdjacent$Names[i] = substr(row.names(XistAdjacent)[i],3,5)
  }

  #Renaming columns 
   colnames(XistTumor)<- c("Expression","Virus", "SexNumeric","SexViralNumeric","Names")
   colnames(XistAdjacent)<- c("Expression","Virus", "SexNumeric","SexViralNumeric","Names")
   
#Creates a box plot of Xist gene expression data of all tumor samples 
  pdf("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_tumor_gene_expression/XIST_tumor_expression_data.pdf", width=12, height=12)
  print(ggplot(XistTumor, aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ xlab("Tumor Samples Expression data") + labs(title="XIST Tumor Samples Expression Data"))
  dev.off() 
  
   pdf("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_tumor_gene_expression/XIST_tumor_expression_500Axis_data.pdf", width=12, height=12)
  print(ggplot(XistTumor, aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2)) + coord_cartesian(ylim=c(0,500))+ xlab("Tumor Samples Expression data") + labs(title="XIST Tumor  Samples Expression Data"))
  dev.off() 
  
  
   pdf("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_adjacent_gene_expression/XIST_tumor_adjacent_expression_data.pdf", width=12, height=12)
  print(ggplot(XistAdjacent, aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ xlab("Adjacent Samples Expression data") + labs(title="XIST Tumor Adjacent Samples Expression Data"))
  dev.off() 
  
   #Creates a box plot of y-linked gene expression data of all tumor adjacent samples
  pdf("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_adjacent_gene_expression/XIST_tumor_adjacent_expression_500Axis_data.pdf", width=12, height=12)
  print(ggplot(XistAdjacent, aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ coord_cartesian(ylim=c(0,500))+ xlab("Adjacent Samples Expression data") + labs(title="XIST Tumor Adjacent Samples Expression Data"))
  dev.off() 
```

**README**
PURPOSE:
This next sections automates a sex check for all the following x degenerate genes using a hash map. The following genes included are
AMELY (tried was not present) 
DDX3Y
EIF1AY
KDM5D
NLGN4Y (tried was not present)
PRKY (tried was not present)
TMSB4Y (tried was not present)
TXLNGY
USP9Y (tried was not pressent)
UTY (KDM6C) (tried was not present)
ZFY (tried was not present)
SRY (tried was not present)
TSPY1
AR

NOTE: TXLNG4Y was not even present on gene card by TXLNGY so I assumed this. 

if gene was not present I check whether or the gene was either not present in original expression data (tumorAdjacentExp) or was filtered out (see next code chunk for my methodology)
```{r x degenerate}

#Storing hashmap as variable
h <- hash()

#Pulling viral etiology designations from metadata
viralType <- metadataSubset$Virus_infection
viralType <- as.data.frame(viralType)

#Renaming the column
colnames(viralType) <- c("Virus")

# Pulling X degenerate expression data from dge variable into individual variables

dge <- as.data.frame(dge)
gene0 <- dge[which(dge$gene_name=="KDM5D"),]
gene1 <- dge[which(dge$gene_name=="DDX3Y"),]
gene2 <- dge[which(dge$gene_name=="EIF1AY"),]
gene3 <- dge[which(dge$gene_name=="TXLNGY"),]
gene4 <- dge[which(dge$gene_name=="TSPY1"),]
gene5 <- dge[which(dge$gene_name=="AR"),]
gene6 <- dge[which(dge$gene_name=="XIST"),]

#outputting XIST expression for later use
write.csv(gene6, "~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/GeneExpressionPlots/data/XISTexpressionData.csv")
#creating data frame of all genes we will be working with
xDegenerateExpression<- rbind(gene0,gene1,gene2,gene3,gene4,gene5,gene6)
#xDegenerateExpression<- t(xDegenerateExpression)
xDegenerateExpression<- as.data.frame(xDegenerateExpression)

#Creating a hashmap that links the ensembl id to gene name of the X degenerate genes so the value is the ensembl id and the key is the gene name (if I ask for gene name it will report the ensembl id)
for (idx in 1:dim(xDegenerateExpression)[1]) {
     name <- xDegenerateExpression$gene_name[idx]
     ensembl_id <- xDegenerateExpression$GENEID[idx]
  if (ensembl_id %in% dge$GENEID) {   
       h[[name]] <- ensembl_id
  }
}

#Creating the for loop so that all of the following steps will be done for each gene located in the hash map
for (gene in keys(h)) {

  #setting the created hashmap into variable 
  ensembl_id <- h[[gene]]
  
  #Pulling expression data from dge variable for the X degenerate genes 
  expressionData <- dge[which(dge$GENEID== ensembl_id),]
  
  #subsetting out all columns that are not expression data
  expressionData <- subset(expressionData, select = -c(chr, start, end, TXNAME, GENEID, length, gene_name))
  expressionData <- t(expressionData)
  expressionData<- as.data.frame(expressionData)
  
  #Combining expression data and viral etiology designations 
  expressionData <- cbind(expressionData,viralType)
  
  #Creating dataframe with the gene expression data and corresponding sample viral etiology 
  #expressionData <- cbind(expressionData,viralType)
  
  #xDegenerateExpression<- as.data.frame(xDegenerateExpression)
  #xDegenerateExpression<- subset(xDegenerateExpression, select = -c(chr, start, end, TXNAME, GENEID, length, gene_name))
  
    #Subsetting by tumor samples
  Tumor_expressionData <- expressionData[grepl("tumor",rownames(expressionData),fixed=TRUE),]
  Tumor_expressionData <- as.data.frame(Tumor_expressionData)
  
  #Adding sample ids as rownames to tumor samples
  rownames(Tumor_expressionData) <- rownames(expressionData)[grepl("tumor",rownames(expressionData),fixed=TRUE)]
  
  #Subsetting by adjacent samples 
  Adjacent_expressionData <- expressionData[grepl("adjacent",rownames(expressionData),fixed=TRUE),]
  Adjacent_expressionData <- as.data.frame(Adjacent_expressionData)
  
  #Adding sample ids as rownames to adjacent samples
  rownames(Adjacent_expressionData) <- rownames(expressionData)[grepl("adjacent",rownames(expressionData),fixed=TRUE)]
  
  #Creating new column that adds a 1 if the sample sex is Male and a 0 if the sample is female
  Tumor_expressionData$SexNumeric <- as.numeric(grepl("XY",rownames(Tumor_expressionData),fixed=TRUE))
  Adjacent_expressionData$SexNumeric <- as.numeric(grepl("XY",rownames(Adjacent_expressionData),fixed=TRUE))
  
  #Renaming columns
  colnames(Tumor_expressionData)<- c("Expression","Virus", "SexNumeric")
  
   #Creating a new column that as a 0 if the sample is male HBV, 1 if male HCV, 2 if female HBV, and 3 if female HCV 
  Tumor_expressionData$SexViralNumeric <- ifelse(Tumor_expressionData$SexNumeric=="1" & Tumor_expressionData$Virus=="HBV", 0,
                                                 ifelse(Tumor_expressionData$SexNumeric=="1" & Tumor_expressionData$Virus=="HCV",1,
                                                        ifelse(Tumor_expressionData$SexNumeric=="0" & Tumor_expressionData$Virus=="HBV",2,3)))
  Adjacent_expressionData$SexViralNumeric <- ifelse(Adjacent_expressionData$SexNumeric=="1" & Adjacent_expressionData$Virus=="HBV", 0,
                                                 ifelse(Adjacent_expressionData$SexNumeric=="1" & Adjacent_expressionData$Virus=="HCV",1,
                                                        ifelse(Adjacent_expressionData$SexNumeric=="0" &  Adjacent_expressionData$Virus=="HBV",2,3)))

  #For loop that adds the sample numbers in a new column called "Names"
  for (i in 1:dim(Tumor_expressionData)[1]) {
    Tumor_expressionData$Names[i] = substr(row.names(Tumor_expressionData)[i],3,5)
  }
  
  for (i in 1:dim(Adjacent_expressionData)[1]) {
    Adjacent_expressionData$Names[i] = substr(row.names(Adjacent_expressionData)[i],3,5)
  }

  #Renaming columns 
  colnames(Tumor_expressionData)<- c("Expression","Virus", "SexNumeric","SexViralNumeric","Names")
  colnames(Adjacent_expressionData)<- c("Expression","Virus", "SexNumeric","SexViralNumeric","Names")
   
  #Creates a box plot of x degenerate gene expression data of all tumor samples 
  pdf(paste("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_tumor_gene_expression/",gene,"_tumor_expression_data.pdf",sep=""), width=12, height=12)
  print(ggplot(Tumor_expressionData, aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ xlab("Tumor Samples Expression data") + labs(title=paste(gene,"Tumor Samples Expression Data")))
  dev.off() 
  
   #Creates a box plot of x degenerate expression data of all tumor  samples with x axis set to 500 to see lowly expressed values
  pdf(paste("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_tumor_gene_expression/",gene,"_tumor_expression_500Axis_data.pdf",sep=""), width=12, height=12)
  print(ggplot(Tumor_expressionData,aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ coord_cartesian(ylim=c(0,500))+xlab("Tumor Samples Expression data") + labs(title=paste(gene,"Tumor Samples Expression Data")))
  dev.off() 
  
  #Creates a box plot of x degenerate gene expression data of all tumor adjacent samples
  pdf(paste("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_adjacent_gene_expression/",gene,"_tumor_adjacent_expression_data.pdf",sep=""), width=12, height=12)
  print(ggplot(Adjacent_expressionData,aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ xlab("Adjacent Samples Expression data") + labs(title=paste(gene,"Tumor Adjacent Samples Expression Data")))
  dev.off() 
  
    #Creates a box plot of x degenerate gene expression data of all tumor adjacent samples  with x axis set to 500 to see lowly expressed values
  pdf(paste("~/Desktop/ResearchProjects/LiverCancer/DEA_removed_samples/newFigures/x_degenerate_adjacent_gene_expression/",gene,"_tumor_adjacent_expression_500Axis_data.pdf",sep=""), width=12, height=12)
  print(ggplot(Adjacent_expressionData,aes(x=reorder(Names,SexViralNumeric),y= Expression,fill=SexViralNumeric)) + geom_bar(stat="identity", colour="black") +theme(axis.text.x = element_text(angle = 90, vjust = 0.75,hjust = 0.75, size=8)) +  scale_x_discrete(guide = guide_axis(n.dodge = 2))+ coord_cartesian(ylim=c(0,500))+xlab("Adjacent Samples Expression data") + labs(title=paste(gene,"Tumor Adjacent Samples Expression Data")))
  dev.off() 

}


```

This shows the process of how I checked the not present genes. First, I figured out if the gene was present in genome used by looking up gene name in "genes" variable. If gene was present I used the ensembl ID given for the next three steps. Second, I checked if genes were present in original data by using the which command in the "tumorAdjacentExp" variable. Third, I check if genes were present in the subsetted variable expression data by using which command in the "tumorAdjacentExpSubset" variable. Finally, I checked if the gene was present by using which command using the ensembl ID instead of gene name. If the gene was found in the original and subsetted expression data and the genome used I said that the gene was not present most likely due to filtering.  
```{r genes not present}
#tried AMELY was not present most likely due to filtering
#gene0 <- dge[which(dge$gene_name=="AMELY"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000099721.13"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000099721.13"),]
#gene0 <- dge[which(dge$GENEID=="ENSG00000099721.13"),]

#tried NLGN4Y was not present most likely due to filtering
#gene3 <- dge[which(dge$gene_name=="NLGN4Y"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000165246.14"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000165246.14"),]
#gene3 <- dge[which(dge$GENEID=="ENSG00000165246.14"),]

#tried PRKYwas not present most likely due to filtering
#gene3 <- dge[which(dge$gene_name=="PRKY"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000099725.14"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000099725.14"),]
#gene3 <- dge[which(dge$GENEID=="ENSG00000099725.14"),]

#tried TMSB4Y was not present most likely due to filtering
#gene3 <- dge[which(dge$gene_name=="TMSB4Y"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000154620.5"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000154620.5"),]
#gene3 <- dge[which(dge$GENEID=="ENSG00000154620.5"),]


#tried USP9Y was not present most likely due to filtering 
#gene4 <- dge[which(dge$gene_name=="USP9Y"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000114374.12"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000114374.12"),]
#gene4 <- dge[which(dge$GENEID=="ENSG00000114374.12"),]

#tried ZFY was not present most likely due to filtering
#gene4 <- dge[which(dge$gene_name=="ZFY"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000067646.11"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000067646.11"),]
#gene4 <- dge[which(dge$GENEID=="ENSG00000067646.11"),]

# tried SRY was not present most likely due to filtering
#gene4 <- dge[which(dge$gene_name=="SRY"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000184895.7"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000184895.7"),]
#gene4 <- dge[which(dge$GENEID=="ENSG00000184895.7"),]

# tried UTY was not present most likely due to filtering, also tried KDM6C did not go by that name
#gene4 <- dge[which(dge$gene_name=="UTY"),]
#check <- tumorAdjacentExp[which(rownames(tumorAdjacentExp)=="ENSG00000183878.15"),]
#check1 <- tumorAdjacentExpSubset[which(rownames(tumorAdjacentExpSubset)=="ENSG00000183878.15"),]
#gene4 <- dge[which(dge$GENEID=="ENSG00000183878.15"),]
```
