---
title: "New MDS Plots"
author: "Annika Jorgensen"
date: "2022-12-12"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# **R Markdown**

Title: "New MDS Plots "  
Author: Annika Jorgensen  
Date: 12/12/2022  
Purpose: This document is to generate additional MDS Plots with different principal components to see the effect of etiology on the data. 

### **Libraries**
The first chunk of code is dedicated to installing the libraries. These libraries are to help execute the differential analysis and helps visualize the data. The code was not included for concision.  
```{r Libraries, include=FALSE}

library(RColorBrewer)
library(matrixStats)
library(ggplot2)
library(edgeR)
library(DESeq2)
library(limma)
library(doParallel)
library(variancePartition)
library(org.Hs.eg.db)
library(clusterProfiler)
library(GOSemSim)
library(biomaRt)
library(UpSetR)
library(VennDiagram)
library(ggrepel)
library(dplyr)
library(stringr)
library(forcats)
library(stats)
library(gridExtra)

setwd('~/Desktop/Research Projects/Liver Cancer Project')
```

**Defining Colors** 

This chunk defines color palette variables that are going to be used in plots later on the script. These variables are defined by converting BrewerCode palettes into palettes that can be used in R. 
```{r Color_Palettes}
viralPalette <- brewer.pal(8, "Set1")
hbvColor <- viralPalette[1]
hcvColor <- viralPalette[2]
bothColor <- viralPalette[3]
neitherColor <- viralPalette[4]

sexTissuePalette <- brewer.pal(12, "Paired")
maleTumorColor <- sexTissuePalette[4]
maleAdjacentColor <- sexTissuePalette[3]
femaleTumorColor <- sexTissuePalette[6]
femaleAdjacentColor <- sexTissuePalette[5]
```

**Read in data**

This code is where you read in all the data files that are going to be used in the script. The data is also converted into a variety of variables that makes the data easier to handle. The data is also cleaned up to make sure the analysis done later is accurate and precise. 
```{r Reading_in_data}
metadata <- read.table("~/Desktop/Research Projects/Liver Cancer Project//Metadata/metadata_for_de.csv", row.names=1,header=TRUE, sep=",") #changing the name of the file
tumorAdjacentExp <- read.table("~/Desktop/Research Projects/Liver Cancer Project/Metadata/japan_all_samples_salmon_expression_counts.txt", row.names = 1, header=TRUE) #changing the name of the file 
colnames(tumorAdjacentExp) <- gsub("\\.", "-", colnames(tumorAdjacentExp)) #changing the column names 

# Importing gene annotations
#genes <- read.table("gencode.v25.chr_patch_hapl_scaff.annotation.bed", header=FALSE, sep="\t")
genes <- read.table("~/Desktop/Research Projects/Liver Cancer Project/Metadata/gencodeTranscripts.txt", header=TRUE, sep="\t")
genes <- data.frame(genes)
tumorAdjacentExp <- tumorAdjacentExp[rownames(tumorAdjacentExp) %in% genes$GENEID ,]
genes <- genes[match(rownames(tumorAdjacentExp), genes$GENEID),]
# Calculating gene length, this is needed for calculating the FPKM values
genes$length <- with(genes, end - start)

# Removing RK023 due to low quality
metadata <- metadata[!(metadata$ID == "RK023") , ]

#Removing both and NBNC samples
metadata <- metadata[!(metadata$Virus_infection == "NBNC"), ]
metadata <- metadata[!(metadata$Virus_infection == "both"), ]

# Subsetting and ordering metadata to match the count matrix
tumorAdjacentExpSubset <- tumorAdjacentExp[,colnames(tumorAdjacentExp) %in% metadata$sampleid]
metadataSubset <- metadata[metadata$sampleid %in% colnames(tumorAdjacentExpSubset),]
metadataSubset <- metadataSubset[match(colnames(tumorAdjacentExpSubset), metadataSubset$sampleid),]
rownames(metadataSubset) <- metadataSubset$sampleid

# Adding tissue type, converting categorical variables to factors
metadataSubset$tumor <- as.numeric(grepl('tumor', metadataSubset$sampleid, ignore.case=T))
metadataSubset$gender_tissue <- paste(metadataSubset$Gender, metadataSubset$tumor, sep="_")
metadataSubset$gender_tissue_viral <- paste(metadataSubset$gender_tissue, metadataSubset$Virus_infection, sep="_")
metadataSubset$library_type <- metadataSubset$strandedness
metadataSubset$library_type <- factor(metadataSubset$library_type)
metadataSubset$tumor <- factor(metadataSubset$tumor)
metadataSubset$Ta <- factor(metadataSubset$Ta)
metadataSubset$Portal_vein_invasion <- factor(metadataSubset$Portal_vein_invasion)
metadataSubset$Hepatic_vein_invasion <- factor(metadataSubset$Hepatic_vein_invasion)
metadataSubset$Bile_duct_invasion <- factor(metadataSubset$Bile_duct_invasion)
metadataSubset$Liver_fibrosisc <- factor(metadataSubset$Liver_fibrosisc)
metadataSubset$Prognosis <- factor(metadataSubset$Prognosis)
```

```{r DGE Object}
# Creating the DGEList object
dge <- DGEList(counts=tumorAdjacentExpSubset, genes=genes)
colnames(dge) <- colnames(tumorAdjacentExpSubset)
dge$samples$sex <- metadataSubset$Gender
dge$samples$viral <- factor(metadataSubset$Virus_infection)
dge$samples$ID <- metadataSubset$ID
dge$samples$tumor <- metadataSubset$tumor
dge$samples$gender_tissue <- metadataSubset$gender_tissue
dge$samples$gender_tissue_viral <- metadataSubset$gender_tissue_viral
dge$samples$library_type <- metadataSubset$library_type
dge$samples$edmonson_grade <- metadataSubset$Edmondson_grade
dge$samples$Ta <- metadataSubset$Ta
dge$samples$survival <- metadataSubset$Overall_survival_month

# Inspecting the N of samples in each group
table(dge$samples$gender_tissue_viral)
```

```{r fpkm}
# ======================================
# Filtering expression data
# ======================================

# Keeping genes that have a mean FPKM of at least 0.5 in at least one of the
# groups under investigation and at least 6 reads in at least 10 samples
fpkm <- rpkm(dge, gene.length=dge$genes$length)

M_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HBV")],1,mean,na.rm=TRUE)
M_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HBV")],1,mean,na.rm=TRUE)
M_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HCV")],1,mean,na.rm=TRUE)
M_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HCV")],1,mean,na.rm=TRUE)

F_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HBV")],1,mean,na.rm=TRUE)
F_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HBV")],1,mean,na.rm=TRUE)
F_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HCV")],1,mean,na.rm=TRUE)
F_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HCV")],1,mean,na.rm=TRUE)

keep <- (M_1_HBV_mean_fpkm > 0.5 | M_0_HBV_mean_fpkm > 0.5 | 
           M_1_HCV_mean_fpkm > 0.5 | M_0_HCV_mean_fpkm > 0.5 |
           F_1_HBV_mean_fpkm > 0.5 | F_0_HBV_mean_fpkm > 0.5 |
           F_1_HCV_mean_fpkm > 0.5 | F_0_HCV_mean_fpkm > 0.5 )

dge <- dge[keep,,keep.lib.sizes=FALSE]
dge <- calcNormFactors(dge, method="TMM")
keep <- rowSums(dge$counts > 6) >= 10
dge <- dge[keep,,keep.lib.size=FALSE]
dge <- calcNormFactors(dge, method="TMM")

# N of genes retained after filtering
dim(dge$genes)
```

```{r Voom}
# ===================================== ======================
# ===========================================================
# Analysis of all tumor vs. tumor-adjacent regardless of sex
# ===========================================================
# ===========================================================

# Creating a new design model matrix with the variable of interest and the
# library type
design <- model.matrix(~0+dge$samples$tumor+dge$samples$library_type+dge$samples$Ta)
colnames(design) <- gsub("dge\\$samples\\$tumor", "tumor", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$library_typeunstranded", "library_type", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta2", "Ta2", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta3", "Ta3", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta4", "Ta4", colnames(design))
head(design)

# Running voom again with the new design matrix.
v <- voomWithQualityWeights(dge, design, plot=TRUE)
```

This creates outputs an MDS Plot that we've previously see showing the dissimilarity between Male and Female Tumor-Tumor adjacent 
```{r Original Sex Stratified MDS Plots}
# ======================================
# Multi-dimensional scaling plot
# ======================================

# MDS plot. The dimensions are in the mds object.

mds <- plotMDS(v, top = 12554, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(v$targets$viral %in% c("HBV"), 17,
                          ifelse(v$targets$viral %in% c("HCV"), 15, 3)),
               col=ifelse(v$targets$gender_tissue=="M_1", "#FFC20A",
                          ifelse(v$targets$gender_tissue=="M_0", "#E66100",
                                 ifelse(v$targets$gender_tissue=="F_1", "#40B0A6",
                                        ifelse(v$targets$gender_tissue=="F_0", "#006CD1", "azure3")))),
               gene.selection = "common") #, gene.selection = "pairwise", labels=tissue,
legend("top", pch=c(15), 
       col=c("#FFC20A", "#E66100", "#40B0A6", "#006CD1"),
       legend=c("M tumor", "M adj", "F tumor", "F adj"))
legend("center", pch=c(17, 15),
       col=c("black"),
       legend=c("HBV", "HCV"))

mds

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/Sex Stratified MDS Plot.pdf")

mds <- plotMDS(v, top = 12554, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(v$targets$viral %in% c("HBV"), 17,
                          ifelse(v$targets$viral %in% c("HCV"), 15, 3)),
               col=ifelse(v$targets$gender_tissue=="M_1", "#FFC20A",
                          ifelse(v$targets$gender_tissue=="M_0", "#E66100",
                                 ifelse(v$targets$gender_tissue=="F_1", "#40B0A6",
                                        ifelse(v$targets$gender_tissue=="F_0", "#006CD1", "azure3")))),
               gene.selection = "common") #, gene.selection = "pairwise", labels=tissue,
legend("top", pch=c(15), 
       col=c("#FFC20A", "#E66100", "#40B0A6", "#006CD1"),
       legend=c("M tumor", "M adj", "F tumor", "F adj"))
legend("center", pch=c(17, 15),
       col=c("black"),
       legend=c("HBV", "HCV"))
dev.off()

```

This code outputs an MDS plot that we have previously seen showing the dissimilarity between the two library types in our dataset
```{r Orginal Library Type MDS Plot}
mds <- plotMDS(v, top = 50, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(v$targets$viral %in% c("HBV"), 17,
                          ifelse(v$targets$viral %in% c("HCV"), 15, 3)),
               col=ifelse(v$targets$library_type=="stranded", "#1A85FF", 
                          ifelse(v$targets$library_type=="unstranded", "#D41159", "black")),
               gene.selection = "common")
legend("top", pch=c(15), 
       col=c("#D41159", "#1A85FF") ,
       legend=c("Unstranded", "Stranded"))
legend("center", pch=c(17, 15),
       col=c("black"),
       legend=c("HBV", "HCV"))
mds

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/Library Types Stratified MDS Plot.pdf")
mds <- plotMDS(v, top = 50, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(v$targets$viral %in% c("HBV"), 17,
                          ifelse(v$targets$viral %in% c("HCV"), 15, 3)),
               col=ifelse(v$targets$library_type=="stranded", "#1A85FF", 
                          ifelse(v$targets$library_type=="unstranded", "#D41159", "black")),
               gene.selection = "common")
legend("top", pch=c(15), 
       col=c("#D41159", "#1A85FF") ,
       legend=c("Unstranded", "Stranded"))
legend("center", pch=c(17, 15),
       col=c("black"),
       legend=c("HBV", "HCV"))

dev.off()
```

This code chunk corrects the library type batch effect and then creates another MDS Plots to check if the correction was made.
```{r Library Type Correction}
# Removing batch effects
vCorrectLibtype <- removeBatchEffect(v, batch=v$targets$library_type)

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/Library Type Correction.pdf")
mds <- plotMDS(vCorrectLibtype, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(v$targets$viral %in% c("HBV"), 17,
                          ifelse(v$targets$viral %in% c("HCV"), 15,
                                 ifelse(v$targets$viral %in% c("both"), 16,  3))),
               col=ifelse(v$targets$gender_tissue=="M_1",
                          "#FFC20A",
                          ifelse(v$targets$gender_tissue=="M_0",
                                 "#E66100",
                                 ifelse(v$targets$gender_tissue=="F_1",
                                        "#40B0A6", "#006CD1"))),
               gene.selection = "common")
legend("bottomleft", pch=c(15), 
       col=c( "#FFC20A", "#40B0A6","#E66100", "#006CD1"),
       legend=c("Male tumor","Female tumor", "Male adjacent","Female adjacent"))
legend("topright", pch=c(17, 15, 16, 3),
       col=c("black"),
       legend=c("HBV", "HCV"))
dev.off()

mds <- plotMDS(vCorrectLibtype, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(v$targets$viral %in% c("HBV"), 17,
                          ifelse(v$targets$viral %in% c("HCV"), 15,
                                 ifelse(v$targets$viral %in% c("both"), 16,  3))),
               col=ifelse(v$targets$gender_tissue=="M_1",
                          "#FFC20A",
                          ifelse(v$targets$gender_tissue=="M_0",
                                 "#E66100",
                                 ifelse(v$targets$gender_tissue=="F_1",
                                        "#40B0A6", "#006CD1"))),
               gene.selection = "common")
legend("bottomleft", pch=c(15), 
       col=c( "#FFC20A", "#40B0A6","#E66100", "#006CD1"),
       legend=c("Male tumor","Female tumor", "Male adjacent","Female adjacent"))
legend("topright", pch=c(17, 15, 16, 3),
       col=c("black"),
       legend=c("HBV", "HCV"))
```
## **New MDS Plots**
This sections will create new MDS plots to see the relationship between sex and etiology in subsets of the Data

```{r Library Correction}
# Removing batch effects
vCorrectLibtype <- removeBatchEffect(v, batch=v$targets$library_type)

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/Library Type Correction Color by strandedness.pdf")
mds <- plotMDS(vCorrectLibtype, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(v$targets$viral %in% c("HBV"), 17,
                          ifelse(v$targets$viral %in% c("HCV"), 15,
                                 ifelse(v$targets$viral %in% c("both"), 16,  3))),
                col=ifelse(v$targets$viral=="HBV", "#1A85FF",
                          ifelse(v$targets$viral=="HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15), 
       col=c("#D41159", "#1A85FF") ,
       legend=c("Unstranded", "Stranded"))
legend("bottomleft", pch=c(17, 15),
       col=c("black"),
       legend=c("HBV", "HCV"))
dev.off()
```
# **Male Tumor**

```{r Male Tumor Data Subset}
#creates variable gendertissueviral that allows us subset vCorrectLibtype by sex and tumor tumor adjacent samples 
vCorrectLibtype <- as.data.frame(vCorrectLibtype)
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male sex and tumor samples 
maletumor<- vCorrectLibtype[, which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HBV")]
maletumor<- as.data.frame(maletumor)

#matching gendertissueviral rownames with column names of maletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maletumor), rownames(gendertissueviral)), ])
```

```{r Male Tumor MDS 12544}

#MDS plot of male tumor samples with all 12,544 differentially expressed genes 
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_tumor_MDS_all_genes_Plot.pdf")
mds <- plotMDS(maletumor, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maletumor, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))


```

# **Male Adjacent**
```{r Male Adjacent Data Subset}
#resetting the variable to not be subsetted by male tumor samples 
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male tumor adjacent samples 
maleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HBV")]
maleadjacent<- as.data.frame(maleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames with column names of maleadjacent
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maleadjacent), rownames(gendertissueviral)), ])
```

```{r Male Adjacent MDS Plots 12554}
#MDS plot of male adjacent samples with all 12,544 differentially expressed genes 
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_adjacent_MDS_all_genes_Plot.pdf")
mds <- plotMDS(maleadjacent, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maleadjacent, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Female Tumor** 
```{r Female Tumor Data Subset}
#resetting gendertissueviral to not be subsetted by male adjacent samples 
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrect Libtype by female tumor samples
femaletumor<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HBV")]
femaletumor<- as.data.frame(femaletumor)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching rownames of gendertissuesviral to column names of femaletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaletumor), rownames(gendertissueviral)), ])

#sets variable to dataframe
#vCorrectLibtype <- as.data.frame(vCorrectLibtype)

#femaletumor <- vCorrectLibtype[,grep("XX", colnames(vCorrectLibtype))]
#femaletumor<-femaletumor[, grep("tumor", colnames(femaletumor))]
```

```{r Female Tumor MDS Plot 12,544}
#MDS plot of female tumor samples with all 12,544 differentially expressed genes
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Tumor_MDS_all_genes_Plot.pdf")
mds <- plotMDS(femaletumor, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaletumor, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

```{r Female Adjacent Data Subset}
#resetting gendertissueviral to not be subsetted by female tumor samples 
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrectLibtype by female adjacent samples
femaleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HBV")]
femaleadjacent<- as.data.frame(femaleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames to femaleadjacent column names 
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaleadjacent), rownames(gendertissueviral)), ])

#vCorrectLibtype <- as.data.frame(vCorrectLibtype)

#femaleadjacent <- vCorrectLibtype[,grep("XX", colnames(vCorrectLibtype))]
#femaleadjacent<-femaleadjacent[, grep("adjacent", colnames(femaleadjacent))]
```

```{r Female Adjacent MDS Plot 12,544}
#MDS plots of female adjacent samples with all 12,544 differentially expressed genes
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Adjacent_MDS_all_genes_Plot.pdf")
mds <- plotMDS(femaleadjacent, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaleadjacent, top = 12554, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Male tumor samples 25 genes**
```{r Male tumor 25 genes MDS plot}
#MDS plot of male tumor samples with the top 25 genes differentially expressed genes 
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_tumor_MDS_25_genes_Plot.pdf")

vCorrectLibtype <- as.data.frame(vCorrectLibtype)
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male sex and tumor samples 
maletumor<- vCorrectLibtype[, which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HBV")]
maletumor<- as.data.frame(maletumor)

#matching gendertissueviral rownames with column names of maletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maletumor), rownames(gendertissueviral)), ])

mds <- plotMDS(maletumor, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maletumor, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))

```
**Male adjacent 25 MDS plot**
```{r Male adjacent 25 genes MDS plot}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male tumor adjacent samples 
maleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HBV")]
maleadjacent<- as.data.frame(maleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames with column names of maleadjacent
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_adjacent_MDS_25_genes_Plot.pdf")
mds <- plotMDS(maleadjacent, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maleadjacent, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```
**Female tumor 25 genes**
```{r female tumor 25 genes}
#resetting gendertissueviral to not be subsetted by male adjacent samples 
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrect Libtype by female tumor samples
femaletumor<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HBV")]
femaletumor<- as.data.frame(femaletumor)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching rownames of gendertissuesviral to column names of femaletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaletumor), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Tumor_MDS_25_genes_Plot.pdf")
mds <- plotMDS(femaletumor, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaletumor, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```
**Female adjacent 25 genes**
```{r female adjacent 25 genes}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrectLibtype by female adjacent samples
femaleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HBV")]
femaleadjacent<- as.data.frame(femaleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames to femaleadjacent column names 
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Adjacent_MDS_25_genes_Plot.pdf")
mds <- plotMDS(femaleadjacent, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaleadjacent, top = 25, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Male tumor 50 genes**
```{r Male tumor 50 genes MDS plot}
#MDS plot of male tumor samples with the top 25 genes differentially expressed genes 
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_tumor_MDS_50_genes_Plot.pdf")

vCorrectLibtype <- as.data.frame(vCorrectLibtype)
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male sex and tumor samples 
maletumor<- vCorrectLibtype[, which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HBV")]
maletumor<- as.data.frame(maletumor)

#matching gendertissueviral rownames with column names of maletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maletumor), rownames(gendertissueviral)), ])

mds <- plotMDS(maletumor, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maletumor, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))

```

**Male adjacent 50 genes**
```{r Male adjacent 50 genes MDS plot}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male tumor adjacent samples 
maleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HBV")]
maleadjacent<- as.data.frame(maleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames with column names of maleadjacent
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_adjacent_MDS_50_genes_Plot.pdf")
mds <- plotMDS(maleadjacent, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maleadjacent, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Female tumor 50 genes**
```{r female tumor 50 genes}
#resetting gendertissueviral to not be subsetted by male adjacent samples 
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrect Libtype by female tumor samples
femaletumor<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HBV")]
femaletumor<- as.data.frame(femaletumor)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching rownames of gendertissuesviral to column names of femaletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaletumor), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Tumor_MDS_50_genes_Plot.pdf")
mds <- plotMDS(femaletumor, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaletumor, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Female adjacent 50 genes**
```{r female adjacent 50 genes}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrectLibtype by female adjacent samples
femaleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HBV")]
femaleadjacent<- as.data.frame(femaleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames to femaleadjacent column names 
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Adjacent_MDS_50_genes_Plot.pdf")
mds <- plotMDS(femaleadjacent, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaleadjacent, top = 50, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Male tumor 100 genes**

```{r Male tumor 100 genes MDS plot}
#MDS plot of male tumor samples with the top 25 genes differentially expressed genes 
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_tumor_MDS_100_genes_Plot.pdf")

vCorrectLibtype <- as.data.frame(vCorrectLibtype)
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male sex and tumor samples 
maletumor<- vCorrectLibtype[, which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HBV")]
maletumor<- as.data.frame(maletumor)

#matching gendertissueviral rownames with column names of maletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maletumor), rownames(gendertissueviral)), ])

mds <- plotMDS(maletumor, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maletumor, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))

```

**Male adjacent 100 genes**
```{r Male adjacent 100 genes MDS plot}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male tumor adjacent samples 
maleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HBV")]
maleadjacent<- as.data.frame(maleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames with column names of maleadjacent
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_adjacent_MDS_100_genes_Plot.pdf")
mds <- plotMDS(maleadjacent, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maleadjacent, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Female tumor 100 genes**
```{r female tumor 100 genes}
#resetting gendertissueviral to not be subsetted by male adjacent samples 
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrect Libtype by female tumor samples
femaletumor<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HBV")]
femaletumor<- as.data.frame(femaletumor)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching rownames of gendertissuesviral to column names of femaletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaletumor), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Tumor_MDS_100_genes_Plot.pdf")
mds <- plotMDS(femaletumor, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaletumor, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Female adjacent 100 genes**
```{r female adjacent 100 genes}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrectLibtype by female adjacent samples
femaleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HBV")]
femaleadjacent<- as.data.frame(femaleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames to femaleadjacent column names 
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Adjacent_MDS_100_genes_Plot.pdf")
mds <- plotMDS(femaleadjacent, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaleadjacent, top = 100, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Male tumor 200 genes**

```{r Male tumor 200 genes MDS plot}
#MDS plot of male tumor samples with the top 25 genes differentially expressed genes 
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_tumor_MDS_200_genes_Plot.pdf")

vCorrectLibtype <- as.data.frame(vCorrectLibtype)
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male sex and tumor samples 
maletumor<- vCorrectLibtype[, which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_1_HBV")]
maletumor<- as.data.frame(maletumor)

#matching gendertissueviral rownames with column names of maletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maletumor), rownames(gendertissueviral)), ])

mds <- plotMDS(maletumor, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maletumor, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_1_HBV", 17,
                          ifelse(gendertissueviral == "M_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topright", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))

```

**Male adjacent 200 genes**
```{r Male adjacent 200 genes MDS plot}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#vCorrectLibtype subsetted by male tumor adjacent samples 
maleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "M_0_HBV")]
maleadjacent<- as.data.frame(maleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames with column names of maleadjacent
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(maleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/M_adjacent_MDS_200_genes_Plot.pdf")
mds <- plotMDS(maleadjacent, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(maleadjacent, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "M_0_HBV", 17,
                          ifelse(gendertissueviral == "M_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="M_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="M_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Female tumor 200 genes**
```{r female tumor 200 genes}
#resetting gendertissueviral to not be subsetted by male adjacent samples 
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrect Libtype by female tumor samples
femaletumor<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_1_HBV")]
femaletumor<- as.data.frame(femaletumor)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching rownames of gendertissuesviral to column names of femaletumor
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaletumor), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Tumor_MDS_200_genes_Plot.pdf")
mds <- plotMDS(femaletumor, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaletumor, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_1_HBV", 17,
                          ifelse(gendertissueviral == "F_1_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_1_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_1_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```

**Female adjacent 200 genes**
```{r female adjacent 200 genes}
gendertissueviral<-as.data.frame( v[["targets"]][["gender_tissue_viral"]])
rownames(gendertissueviral) <- colnames(vCorrectLibtype)

#subsetting vCorrectLibtype by female adjacent samples
femaleadjacent<- vCorrectLibtype[ , which(gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HCV" | gendertissueviral$`v[["targets"]][["gender_tissue_viral"]]` == "F_0_HBV")]
femaleadjacent<- as.data.frame(femaleadjacent)

#subsets by male sex
#maleadjacent <- vCorrectLibtype[, grep("XY", colnames(vCorrectLibtype))]
#subsets by tumor adjacent
#maleadjacent <- maleadjacent[, grep("adjacent", colnames(maleadjacent))]

#matching gendertissueviral rownames to femaleadjacent column names 
gendertissueviral <- as.data.frame(gendertissueviral[match(colnames(femaleadjacent), rownames(gendertissueviral)), ])

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/MDS Plots/F_Adjacent_MDS_200_genes_Plot.pdf")
mds <- plotMDS(femaleadjacent, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
dev.off()

mds <- plotMDS(femaleadjacent, top = 200, ndim = 10, dim.plot = c(1,2), plot=TRUE, cex=2,
               pch=ifelse(gendertissueviral == "F_0_HBV", 17,
                          ifelse(gendertissueviral == "F_0_HCV", 15, 3)),
               col=ifelse(gendertissueviral =="F_0_HBV", "#1A85FF",
                          ifelse(gendertissueviral =="F_0_HCV", "#D41159","black")),
               gene.selection = "common")
legend("topleft", pch=c(15, 17),
        col=c("#D41159", "#1A85FF"),
       legend=c("HCV", "HBV"))
```