---
title: "DEG_Analysis Tumor vs. Tumor adjacent."
output: html_document
Author: Annika Jorgensen
Date: 05/05/2022
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# **R Markdown**

Title: "DEG_Analysis Tumor vs. Tumor"  
Author: Annika Jorgensen  
Date: 02/02/2022  
Purpose: This document is for the author to parse out the tumor vs. tumor adjacent code from the "DEG_changed_comparison" file as well as demonstrate understanding of the code and theory.

### **Libraries**

The first chunk of code is dedicated to installing the libraries. These libraries are to help execute the differential analysis and helps visualize the data. The code was not included for concision.  
```{r Libraries, include=FALSE}

library(RColorBrewer)
library(matrixStats)
library(ggplot2)
library(edgeR)
library(DESeq2)
library(limma)
library(doParallel)
library(variancePartition)
library(org.Hs.eg.db)
library(clusterProfiler)
library(GOSemSim)
library(biomaRt)
library(UpSetR)
library(VennDiagram)
library(ggrepel)
library(dplyr)
library(stringr)
library(forcats)
library(stats)
library(gridExtra)
```

## **Environment parameters**

This next section of code is dedicated to the environmental parameters. Environmental parameters are a series of variables and other code that will help make the rest of the script be easier to make and run later on. 

**Working Directory** 

A working directory is a code that iterates a file path on your computer that sets where the default location of any files that you read into R. Working directories work different in R files than R Markdowns. R Markdown files require directories to be defined at the end of each code chunk. Meaning from here on out you will see working directories being defined at the end of each code chunk. 
```{r Working_Directory}
setwd('~/Desktop/Research Projects/Liver Cancer Project')
```

**Defining Colors** 

This chunk defines color palette variables that are going to be used in plots later on the script. These variables are defined by converting BrewerCode palettes into palettes that can be used in R. 
```{r Color_Palettes}
viralPalette <- brewer.pal(8, "Set1")
hbvColor <- viralPalette[1]
hcvColor <- viralPalette[2]
bothColor <- viralPalette[3]
neitherColor <- viralPalette[4]

sexTissuePalette <- brewer.pal(12, "Paired")
maleTumorColor <- sexTissuePalette[4]
maleAdjacentColor <- sexTissuePalette[3]
femaleTumorColor <- sexTissuePalette[6]
femaleAdjacentColor <- sexTissuePalette[5]
```

**Read in data**

This code is where you read in all the data files that are going to be used in the script. The data is also converted into a variety of variables that makes the data easier to handle. The data is also cleaned up to make sure the analysis done later is accurate and precise. 
```{r Reading_in_data}
metadata <- read.table("~/Desktop/Research Projects/Liver Cancer Project/Metadata/metadata_for_de.csv", row.names=1,header=TRUE, sep=",") #changing the name of the file
tumorAdjacentExp <- read.table("~/Desktop/Research Projects/Liver Cancer Project/Metadata/japan_all_samples_salmon_expression_counts.txt", row.names = 1, header=TRUE) #changing the name of the file 
colnames(tumorAdjacentExp) <- gsub("\\.", "-", colnames(tumorAdjacentExp)) #changing the column names 
```

**Gene Length**

This next code chunk is very similar However, it does calculate **gene length** which is the done by first defining a variable named  "gene"" and then changing the data type to a data frame. You then redefine "tumorAdjacentExp" (defined above) to have the rows of the previous "tumorAdjacentExp" and then the columns of "GENEID"" that lies within "gene".  
Gene length is then defined to have "width" of genes in the rows and 'end-start' as a column

**Running identical function**
We ran the identical function to see if the inputs of the match function are of length one. The function outputted a true value therefore they are identical. 
```{r Gene_length}
genes <- read.table("~/Desktop/Research Projects/Liver Cancer Project/Metadata/gencodeTranscripts.txt", header=TRUE, sep="\t")
genes <- data.frame(genes)
tumorAdjacentExp <- tumorAdjacentExp[rownames(tumorAdjacentExp) %in% genes$GENEID ,]
#genes <- genes[match(rownames(tumorAdjacentExp), genes$GENEID),]

genes <- genes[match(rownames(tumorAdjacentExp), genes$GENEID),]

identical(rownames(tumorAdjacentExp),genes$GENEID)

# Calculating gene length, this is needed for calculating the FPKM values
genes$length <- with(genes, end - start)
```

**Low Quality**

The next line shows a sample being removed due to low quality. 
```{r Low_Quality}
metadata<-metadata[!(metadata$ID == "RK023"), ]
```

**Subsetting data**

This next chunk of data is dedicated to sub-setting and organizing the data to make it easier to use going forward. Sub-setting means that the data is being organized to match a count matrix. In this specific case the count matrix is the sample ID attached to the tumors.

**Running identical function**

We ran the identical function to see if the columns are of length one in the match function. The function returned a TRUE value therefore they are of proper length. 
```{r Subsetting}
tumorAdjacentExpSubset <- tumorAdjacentExp[,colnames(tumorAdjacentExp) %in% metadata$sampleid]
metadataSubset <- metadata[metadata$sampleid %in% colnames(tumorAdjacentExpSubset),]
metadataSubset <- metadataSubset[match(colnames(tumorAdjacentExpSubset), metadataSubset$sampleid),]
identical(colnames(tumorAdjacentExpSubset),metadataSubset$sampleid)
rownames(metadataSubset) <- metadataSubset$sampleid
```

**Tissue object**

This next chunk of data is taking the meta data and subsetting it in such a way that converts a series of categorical variables into factors. This data also adds a tissue type. 
```{r Tissue}
metadataSubset$tumor <- as.numeric(grepl('tumor', metadataSubset$sampleid, ignore.case=T))
metadataSubset$gender_tissue <- paste(metadataSubset$Gender, metadataSubset$tumor, sep="_")
metadataSubset$gender_tissue_viral <- paste(metadataSubset$gender_tissue, metadataSubset$Virus_infection, sep="_")
metadataSubset$library_type <- metadataSubset$strandedness
metadataSubset$library_type <- factor(metadataSubset$library_type)
metadataSubset$tumor <- factor(metadataSubset$tumor)
metadataSubset$Ta <- factor(metadataSubset$Ta)
metadataSubset$Portal_vein_invasion <- factor(metadataSubset$Portal_vein_invasion)
metadataSubset$Hepatic_vein_invasion <- factor(metadataSubset$Hepatic_vein_invasion)
metadataSubset$Bile_duct_invasion <- factor(metadataSubset$Bile_duct_invasion)
metadataSubset$Liver_fibrosisc <- factor(metadataSubset$Liver_fibrosisc)
metadataSubset$Prognosis <- factor(metadataSubset$Prognosis)
```

**Creating DGE Object**

This next chunk of code creates something called a DGEList Object. This object contains a dataset that is to be analyzed later in the script. Specifically the object contains:  
1. Counts-- numeric matrix containing read counts    
2. group-- vector giving the experimental conditiona for each sample  
3. genes-- data frame information for the genes for which we have count data   
4. remove.zeros-- whether to remove rows that have 0 total count 

The last line of code takes the amount of samples and places them into a table for easy read and inspection.
```{r DGE object}
dge <- DGEList(counts=tumorAdjacentExpSubset, genes=genes)
colnames(dge) <- colnames(tumorAdjacentExpSubset)
dge$samples$sex <- metadataSubset$Gender
dge$samples$viral <- factor(metadataSubset$Virus_infection)
dge$samples$ID <- metadataSubset$ID
dge$samples$tumor <- metadataSubset$tumor
dge$samples$gender_tissue <- metadataSubset$gender_tissue
dge$samples$gender_tissue_viral <- metadataSubset$gender_tissue_viral
dge$samples$library_type <- metadataSubset$library_type
dge$samples$edmonson_grade <- metadataSubset$Edmondson_grade
dge$samples$Ta <- metadataSubset$Ta
dge$samples$survival <- metadataSubset$Overall_survival_month

##======================

#Inspecting the N of samples in each group

table(dge$samples$gender_tissue_viral)
```

**Calculating fpkm values **

This chunk of code takes the fpkm of all the genes in the dataset and calculates the mean. They also filter out genes that have a fpkm of 0.5 or lower. 

Fpkm stands for  fragments per kilo base of exon per million this term is interchangable with Rpkm (reads per kilobase of exon per million. This measure is a normalization method which allows us to compare gene expression levels by rescaling both library size and gene length.   

Fpkm is calculated by multiplying the number of reads mapped to a gene by 1,000,  1,000,000 and then dividing that number by the total number of mapped reads to gene length in base pairs. 

Please note that that calculation is done for RPKM which is analogous to Fpkm.
```{r Calculating fpkm}
# ======================================
# Filtering expression data
# ======================================

# Keeping genes that have a mean FPKM of at least 0.5 in at least one of the
# groups under investigation and at least 6 reads in at least 10 samples
fpkm <- rpkm(dge, gene.length=dge$genes$length)
```

Here the fpkm is calculated from all the various tissue samples, are filtered for greater than 0.5 and put into a variable named "keep" which a cutoff point that is going to be used in limma/voom analysis 
```{r fpkm tissues}

M_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HBV")],1,mean,na.rm=TRUE)
M_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HBV")],1,mean,na.rm=TRUE)
M_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_HCV")],1,mean,na.rm=TRUE)
M_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_HCV")],1,mean,na.rm=TRUE)
M_1_HBVHCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_both")],1,mean,na.rm=TRUE)
M_0_HBVHCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_both")],1,mean,na.rm=TRUE)
M_1_NBNC_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_1_NBNC")],1,mean,na.rm=TRUE)
M_0_NBNC_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="M_0_NBNC")],1,mean,na.rm=TRUE)

F_1_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HBV")],1,mean,na.rm=TRUE)
F_0_HBV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HBV")],1,mean,na.rm=TRUE)
F_1_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_HCV")],1,mean,na.rm=TRUE)
F_0_HCV_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_HCV")],1,mean,na.rm=TRUE)
F_1_NBNC_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_1_NBNC")],1,mean,na.rm=TRUE)
F_0_NBNC_mean_fpkm <- apply(as.data.frame(fpkm)[(dge$samples$gender_tissue_viral=="F_0_NBNC")],1,mean,na.rm=TRUE)

keep <- (M_1_HBV_mean_fpkm > 0.5 | M_0_HBV_mean_fpkm > 0.5 | 
           M_1_HCV_mean_fpkm > 0.5 | M_0_HCV_mean_fpkm > 0.5 |
           M_1_HBVHCV_mean_fpkm > 0.5 | M_0_HBVHCV_mean_fpkm > 0.5 |
           M_1_NBNC_mean_fpkm > 0.5 | M_0_NBNC_mean_fpkm > 0.5 |
           F_1_HBV_mean_fpkm > 0.5 | F_0_HBV_mean_fpkm > 0.5 |
           F_1_HCV_mean_fpkm > 0.5 | F_0_HCV_mean_fpkm > 0.5 |
           F_1_NBNC_mean_fpkm > 0.5 | F_0_NBNC_mean_fpkm > 0.5)

```

**DGE object organization**

This chunk is further organizes and counts the libraries to be more tangible for later on as well as calculates the normalization factors (not normalizing the data) to use later on in the limma/voom DEG.

The normalization factors are calculated using Trimmed Mean of M-values (TMM). TMM is a between sample normalization that assumes that most genes are not differentially expressed. TMM normalizes the total RNA output among the samples, not considering gene length nor library size. TMM also considers the RNA population which makes it effective with samples that have diverse RNA repertoires.    

TMM takes the library size normalized read count for each gene in each sample and calculates the log2 fold change between two samples (M-value). From there you calculate the absolute expression count (A values) which is the sum of the log2 fold change of treated sample count plus the log2 fold change of the control sample count divided by two.  

M-values and A-values are double trimmed by 30% and 5% respectively. You then get the weight mean M after trimming and calculate the normalization factor.  
```{r DGE again}
dge <- dge[keep,,keep.lib.sizes=FALSE]
dge <- calcNormFactors(dge, method="TMM")
keep <- rowSums(dge$counts > 6) >= 10
dge <- dge[keep,,keep.lib.size=FALSE]
dge <- calcNormFactors(dge, method="TMM")
```

**Counting number of FPKM genes**

This code counts the number of genes that make it past the cutoff point.  
```{r Number of Genes}

# N of genes retained after filtering
dim(dge$genes)
```
## **DGE Analysis tumor vs. tumor- adjacent**

This section is doing voom/limma and Go/KEGG analysis on all of the tumor vs. tumor adajcent samples 

**Design matrix**

This code chunk is creating a design matrix to analyze tumor vs. tumor adjacent regardless of sex. This design matrix creates a model matrix that takes the inputs that we want to consider for later limma/voom analysis.   

In this specific design matrix we are considering all tumor samples and all tumor adjacent samples.  

The colnames code is just renaming columns of the design matrix to identify what they are for easier readability. 
```{r First Design Matrix}
# ===========================================================
# ===========================================================
# Analysis of all tumor vs. tumor-adjacent regardless of sex
# ===========================================================
# ===========================================================

# Creating a new design model matrix with the variable of interest and the
# library type
design <- model.matrix(~0+dge$samples$tumor+dge$samples$library_type+dge$samples$Ta)

#I just put design matrix in twice because I didn't remember what inputs to put in. 
identical(design,design,num.eq=TRUE)

colnames(design) <- gsub("dge\\$samples\\$tumor", "tumor", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$library_typeunstranded", "library_type", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta2", "Ta2", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta3", "Ta3", colnames(design))
colnames(design) <- gsub("dge\\$samples\\$Ta4", "Ta4", colnames(design))
head(design)
```

**voom**

voom is a function that lies within a package called limma. limma/voom is used in DGE analysis. voom is a function that takes the counts in a metadata set and transforms them into log2 of TMM values calculated above in the normalization factors. A linear model is then fitted to the TMM for each gene and residuals are calculated. A smoothed curve is then fitted to the square root of the residual standard deviation by the average expression (this is the red line). This smooth curve is then used to obtain weights for each gene and sample that are passed into limma along with the TMM values.  
```{r First voom run}
# Running voom again with the new design matrix.

#pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/tumor tumor adjacent voom.pdf", width=12, height=12)
v <- voomWithQualityWeights(dge, design, plot=TRUE)
dev.off()
```

**limma**

This sections marks the beginning of running limma. limma creates a linear fit to the data, makes comparisons of the fitted data, and applies Bayes smoothing. limma startes by creating a variable that has all of the duplicate correlation values on v and design. These correlation values will be used later in a linear fit.
```{r Limma fit tumor v. tumor- adjacent}
# ============================================================================================
# Differential expression analysis with limma - all male tumor adjacent vs. non-tumor-adjacent
# ===========================================================================================

# Block design for individual. This is used in tumor-normal comparisons with
# paired samples.
corfit <- duplicateCorrelation(v, design, block = v$targets$ID)
# This should give a positive correlation value. It represents the
# correlation between measurements made on the same person.
corfit$consensus
```

**limma graph**

This is the linear model with limma, notice that the correlation values were the duplicate correlations used earlier
```{r Linear fit limma}
# Fitting the linear model with limma.
# If using paired samples, the within-patient correlation and a block design
# for patient is used to account for pairwise samples
fit <- lmFit(v, design, block = v$targets$ID, correlation = corfit$consensus)
```

**Coefficient vector**

This code chunk involves extracting coefficients from the linear fit model and storing them in a vector for later use.
```{r Design Matrix}
# Contrast design for differential expression
# Defining pairwise comparisons
contrasts <- makeContrasts(Adjacent_vs_Tumor = tumor1 - tumor0,
                           levels=colnames(design))
head(contrasts)
# Assigning all comparisons to a vector for later
allComparisons <- colnames(contrasts)
```

**Constrast Analysis**

This next code chunk reorients the linear model obtained earlier and obtains the coefficients and standard errors from the model. This step also sets us up to apply Empirical Bayes smoothing. 
```{r Contrast Analysis}
# Running contrast analysis
vfit <- contrasts.fit(fit, contrasts = contrasts)
# Look at N of DEGs with adj. p <0.01 and log2FC>2
summary(decideTests(vfit, adjust.method = "BH", p.value = 0.05, lfc = 2))
```

**Bayes smoothing**

This code chunk uses Empirical Bayes smoothing to plot the final model after doing the limma and voom analysis. 

Empirical Bayes smoothing is a way to account for uncertainty. The technique uses the population in a region as a measure of confidence. Meaning that areas with low margin of error are left untouched while estimates with higher margin of error are moved closer to the global average.  
```{r Bayes Analysis}
# Computing differential expression based on the empirical Bayes moderation of
# the standard errors towards a common value. Robust = should the estimation of
# the empirical Bayes prior parameters be robustified against outlier sample
# variances?
veBayesFit <- eBayes(vfit, robust=TRUE)
#pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/tumor tumor adjacent final model plot.pdf", width=12, height=12)
plotSA(veBayesFit, main = "Final model: Mean-variance trend")
dev.off()
vTopTable <- topTable(veBayesFit, n=Inf, p.value=1, lfc=0)
DEGs <- topTable(veBayesFit, n=Inf, p.value=0.05, lfc=2)
DEGs_print <- data.frame(DEGs$GENEID, DEGs$gene_name, DEGs$adj.P.Val, DEGs$logFC)
write.csv(DEGs_print, "~/Desktop/Research Projects/Liver Cancer Project/Gene lists/gene_list_tumor_vs_tumor_adjacent.csv")
```

## **Data Visualization**

This section creates Volcano plots for easy visualization of the gene lists.

**Volcano plot object**

This code chunk involves setting up the variables and plot setting necessary to create a Volcano Plot of Tumor vs. Tumor-adjacent samples.  
```{r Constructing Volcano Plot}
# ===========================================
#Volcano plot of tumor vs tumor-adjacent
# ===========================================
df <- data.frame(vTopTable$adj.P.Val, vTopTable$logFC, vTopTable$chr, vTopTable$GENEID, vTopTable$gene_name)
colnames(df) <- c("adj.P.Val", "logFC", "chr", "id", "name")
dfSig <- df[(abs(df$logFC) >= 2 & df$adj.P.Val <= 0.05),]$id


dfAnons <- subset(df, chr != "chrX" & chr != "chrY" & !(id %in% dfSig))
dfAnons <- cbind(dfAnons , rep(1, nrow(dfAnons)))
colnames(dfAnons)[6] <- "Color"

dfXnons <- subset(df, chr == "chrX" & !(id %in% dfSig))
dfXnons <- cbind(dfXnons, rep(2, nrow(dfXnons)))
colnames(dfXnons)[6] <- "Color"

dfYnons <- subset(df, chr == "chrY" & !(id %in% dfSig))
dfYnons <- cbind(dfYnons, rep(3, nrow(dfYnons)))
colnames(dfYnons)[6] <- "Color"

dfA <- subset(df, chr != "chrX" & chr != "chrY" & id %in% dfSig)
dfA <- cbind(dfA, rep(4, nrow(dfA)))
colnames(dfA)[6] <- "Color"

dfX <- subset(df, chr == "chrX" & id %in% dfSig)
dfX <- cbind(dfX, rep(5, nrow(dfX)))
colnames(dfX)[6] <- "Color"

dfY <- subset(df, chr == "chrY" & id %in% dfSig)
dfY <- cbind(dfY, rep(6, nrow(dfY)))
colnames(dfY)[6] <- "Color"

dfPlot <- rbind(dfAnons, dfXnons, dfYnons, dfA, dfX, dfY)
dfPlot$Color <- as.factor(dfPlot$Color)
```

**Volcano plot without significance levels**

This code chunk constructs the plot object of Tumor vs. Tumor adjacent volcano plot and customizes the plot. Volcano plots are a type of scatter that is used to quickly identify changes in large data sets. Volcano plots combine a statistical significance level with a magnitude of the change to allow quick visual identification of data points that display a large magnitude change that is statistically significant.  
```{r Constructing Plot Object}
# Constructing the plot object. The colors will not work the way they should if
# one of the groups doesn't have any genes
p <- ggplot(data = dfPlot, aes(x = logFC, y = -log10(adj.P.Val), color=Color )) +
  geom_point(alpha = 0.5, size = 8) +
  theme_bw() +
  theme(legend.position = "none") +
  xlim(c(-15, 15)) + ylim(c(0, 130)) +
  scale_color_manual(values = c("azure3", "pink", "seagreen2", "black", "mediumvioletred", "springgreen")) +
  labs(x=expression(log[2](FC)),
       y=expression(-log[10] ~ "(FDR-adjusted " ~ italic("p") ~ "-value)")) +
  theme(axis.title.x=element_text(size=12), 
        axis.text.x=element_text(size=10)) +
  theme(axis.title.y=element_text(size=12),
        axis.text.y=element_text(size=10))
p

forLabel <- subset(dfPlot, adj.P.Val<=0.05 & abs(logFC)>=2)
```

**Volcano plot with significance levels**

This next code chunk adds specific significance thresholds to the volcano plot. The significance thresholds can be seen in the dotted lines. 
```{r Significance thresholds RMD}
# Adding lines for significance thresholds
#pdf("~/3.0 Hasting Research/Liver Cancer Analysis/DEG Figures/Volcano_plot_tumor_tumoradjacent_DEGs.pdf", width=12, height=12)

#plotting to RMD file
p + geom_hline(yintercept = 1.3, colour="#000000", linetype="dashed"
) + geom_vline(xintercept = 2, colour="#000000", linetype="dashed"
) + geom_vline(xintercept = -2, colour="#000000", linetype="dashed"
) + geom_text_repel(data=forLabel, max.iter=1000, box.padding = 0.25, force=1, aes(x = logFC, y = -log10(adj.P.Val), label=name, color=Color), size=8)
```


```{r Significance thresholds to PDF}
#plotting to PDF
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/Volcanoplots/Volcano_Tumor_vs_Tumor_adjacent.pdf", width=12, height=12)
p + geom_hline(yintercept = 1.3, colour="#000000", linetype="dashed"
) + geom_vline(xintercept = 2, colour="#000000", linetype="dashed"
) + geom_vline(xintercept = -2, colour="#000000", linetype="dashed"
) + geom_text_repel(data=forLabel, max.iter=1000, box.padding = 0.25, force=1, aes(x = logFC, y = -log10(adj.P.Val), label=name, color=Color), size=8)
```
```{r new volcano plot}
df <- data.frame(vTopTable$adj.P.Val, vTopTable$logFC, vTopTable$chr, vTopTable$GENEID, vTopTable$gene_name)
colnames(df) <- c("adj.P.Val", "logFC", "chr", "id", "name")
dfSig <- df[(abs(df$logFC) >= 2 & df$adj.P.Val <= 0.05),]$id

p<- ggplot(data=df, aes(x=logFC, y=-log10(adj.P.Val))) + geom_point(alpha=0.5)+theme_bw()

p2 <-  p+ geom_vline(xintercept=c(-2, 2), linetype= "dashed", size=1.5) + geom_hline(yintercept= 2, linetype= "dashed", size=1.5) +  xlim(c(-10, 10)) #+ ylim(c(0, 50))

p3<- p2+ theme(legend.position = c(0.925,0.75)) + geom_point(aes(color=abs(logFC)>=2 & adj.P.Val<=0.05), size=4)+theme(legend.position="none")+scale_color_manual(values = c("black","maroon"))

p4<- p3+theme(axis.title = element_text(size=35), axis.text = element_text(size=35, color="black"), legend.title = element_text(size=35)) + labs(x= "log fold change", y="-log(FDR- adjusted p value)")

p4

pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/Volcanoplots/Volcano_Tumor_vs_Tumor_adjacent.pdf", width=12, height=12)
p4
dev.off()
```
## **Results**

This sections shows the results of the DEG and Go/KEGG analysis 

**DEG results**

This code chunk shows the results of our DEG analysis by retrieving specific data from a package called biomart. The combination of the DEG analysis results and the data from biomart is printed into a table.

The label "kegg-enzyme" was not valid so it has been deleted.
```{r DEGs results}
# =================
# Enriched pathways
# =================

degResult_genes <- DEGs
degResult_genes$hgnc_symbol <- degResult_genes$name
ensembl <- useEnsembl(biomart="ensembl", dataset="hsapiens_gene_ensembl")
#biomart <- getBM(attributes=c('ensembl_gene_id', 'hgnc_symbol', 'entrezgene_id', 'kegg_enzyme'), filters = 'hgnc_symbol', values = degResult_genes$gene_name, mart = ensembl)
biomart <- getBM(attributes=c('ensembl_gene_id', 'hgnc_symbol', 'entrezgene_id'), filters = 'hgnc_symbol', values = degResult_genes$gene_name, mart = ensembl)
colnames(degResult_genes) <- c("chr", "start", "end", "TXNAME", "GENEID", "hgnc_symbol", "length", 
                               "logFC", "AveExpr", "t", "P.Value", "adj.P.Val", "B")
degResult_genes_biomart <- merge(degResult_genes, biomart, by="hgnc_symbol")
head(degResult_genes_biomart)
degResult <- DEGs
degResult$hgnc_symbol <- degResult$gene_name
degResult_biomart <- merge(degResult, biomart, by="hgnc_symbol")
degResult_biomart <- degResult_biomart[complete.cases(degResult_biomart), ]
```

**GO Analysis**

This code shows the dot plot of a GO enrichment analysis. GO enrichment analysis uses "Gene Ontology" system of classification to put genes into predefined "bins" allows high throughput experiments where certain data can be retrieved. GO is used to see which GO terms appear more frequently then by chance.  
```{r Go analysis}
geneList <- degResult_biomart$entrezgene_id
geneList_quant <- degResult_biomart$logFC
ego <- enrichGO(geneList, OrgDb = "org.Hs.eg.db", ont="BP", readable=TRUE)
#pdf("~/3.0 Hasting Research/Liver Cancer Analysis/GO_KEGG Figures/dotplot_all_tumor_vs_tumor_adjacent.pdf", width=12, height=12)

dotplot(ego, showCategory=30)
```

```{r GO analysis to PDF}
geneList <- degResult_biomart$entrezgene_id
geneList_quant <- degResult_biomart$logFC
ego <- enrichGO(geneList, OrgDb = "org.Hs.eg.db", ont="BP", readable=TRUE)
#pdf("~/3.0 Hasting Research/Liver Cancer Analysis/GO_KEGG Figures/dotplot_all_tumor_vs_tumor_adjacent.pdf", width=12, height=12)
pdf("~/Desktop/Research Projects/Liver Cancer Project/Figures/Gokeggplots/GO_KEGG_Tumor_vs_Tumor_adjacent.pdf", width=12, height=12)
print(dotplot(ego, showCategory=30))
#dev.off()
```

